### PR TITLE
Reading and processing android sensor data using python with CSV read()

### LINKED ISSUE #6159: Reading and processing android sensor data using python with CSV read
### Topic Suggestion
Reading and processing android sensor data using python with CSV read

#### Introduction
Reading is an action performed by computers to acquire data from a source and place it into their volatile memory for processing. Computers may read information from various sources, such as the Internet or audio and video input ports. Data processing is a process of converting raw data into meaningful information. It involves the digestion of the raw data into a more reasonable form.

This tutorial explains how one records data of various smartphone sensors and processes that data offline using Python. First, you extract the sensor's data, the CSV file is read by Python and the extracted data is processed for display. Also, it is processed for taking wavelet transform.

#### Key takeaways 
- Various mobile sensors.
- Why record and how to record mobile sensor data.
- Third party android app to collect sensor data.
- Python code using csv.reader to read Accelerometer data

#### Article quality
This article shows how you can collect data using your android phone and process this data in python. It also shows how you can link a third party android app to python and share this data. 

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by josephodhis**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #6159 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by srishilesh** [comment]
Hello @josephodhis ,

Thanks for submitting this PR.

Your PR has entered our reviewing queue. Please wait until a peer-reviewer pings you to get started.

Meanwhile, we suggest you to go through existing approved PRs/published articles to see if you have missed out anything.

Thanks.

File: content/articles/reading-and-processing-android-sensor-data-using-python-with-csv-read/index.md
Comment (paulodhiambo) [inline content/articles/reading-and-processing-android-sensor-data-using-python-with-csv-read/index.md:35]: Add the image extension for this image

**Review by paulodhiambo — CHANGES_REQUESTED** [review]
Amazing article, please add a link to complete source code.

**Comment by louisefindlay23** [comment]
Plagiarism check complete

**Comment by josephodhis** [comment]
@paulodhiambo changes made.

File: content/articles/reading-and-processing-android-sensor-data-using-python-with-csv-read/index.md
Comment (Kayere) [inline content/articles/reading-and-processing-android-sensor-data-using-python-with-csv-read/index.md:93]: Maximum header size allowed is header three.

File: content/articles/reading-and-processing-android-sensor-data-using-python-with-csv-read/index.md
Comment (Kayere) [inline content/articles/reading-and-processing-android-sensor-data-using-python-with-csv-read/index.md:95]: Please add backticks to your code snippets.

File: content/articles/reading-and-processing-android-sensor-data-using-python-with-csv-read/index.md
Comment (Kayere) [inline content/articles/reading-and-processing-android-sensor-data-using-python-with-csv-read/index.md:140]: Change this header size too.

**Review by Kayere — CHANGES_REQUESTED** [review]
Hi @josephodhis, please go over the changes I have requested so I can continue with the final review.

**Comment by paulodhiambo** [comment]
Hello @Kayere I made the corrections, didn't notice the contributor made some changes after completing tech review.

**Comment by Kayere** [comment]
Hi @josephodhis, I am done with the final review. Just one request before I merge this PR. Please change your author avatar - upload a professional picture.

**Comment by josephodhis** [comment]
@Kayere i have uploaded a new avatar.

**Pull request merged**
